SilKit/source/core/vasio/RingBuffer.cpp

```diff
@@ -0,0 +1,173 @@
+// SPDX-FileCopyrightText: 2023 Vector Informatik GmbH
+//
+// SPDX-License-Identifier: MIT
+
+#include "RingBuffer.hpp"
+
+#include <algorithm>
+#include <cstring>
+
+#include "silkit/participant/exception.hpp"
+
+namespace SilKit {
+namespace Core {
+
+RingBuffer::RingBuffer(size_t capacity)
+{
+    _buffer.resize(capacity);
+}
+
+void RingBuffer::AdvanceWPos(size_t numBytes)
+{
+    _wPos = (_wPos + numBytes) % Capacity();
+    _size += numBytes;
+
+    SizeCheck();
+}
+
+void RingBuffer::AdvanceRPos(size_t numBytes)
+{
+    _rPos = (_rPos + numBytes) % Capacity();
+    _size -= numBytes;
+
+    SizeCheck();
+}
+
+bool RingBuffer::Peek(std::vector<uint8_t>& elem) const
+{
+    // make sure, we only copy as many bytes as are contained in the buffer
+    if (elem.size() > _size)
+    {
+        return false;
+    }
+
+    // copy data from first contiguous array (of occupied memory)
+    size_t numBytesArrayOne = std::min((Capacity() - _rPos), elem.size());
+    memcpy(elem.data(), _buffer.data() + _rPos, numBytesArrayOne);
```

Comment:
Not entirely sure why the `std::` namespace was removed.

Afaik the C++ standard does not guarantee that the 'C' names (i.e., in the global namespace) are actually available, it just doesn't forbid it. See https://stackoverflow.com/a/41791540 for a quote from the standard.

I suppose that there are very few compilers that don't provide the functions in the global namespace though.

Reply:
This can also just stay as-is as long as MSVC is happy (since I've seen it choke on various functions from `<cstring>` before).

See the `Important` note in [the msdn article on memcpy](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/memcpy-wmemcpy?view=msvc-170#remarks) for the (non-standard) "deprecation" under some circumstances.

Reply:
Done. Thanks for the hint and explanations!